src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java

```diff
@@ -419,4 +471,14 @@ public AttachToCaseEventData(
         merged.putAll(modifiedFields);
         return merged;
     }
+
+    private ServiceConfigItem getServiceConfig(CaseDetails caseDetails) {
+        return hasServiceNameInCaseTypeId(caseDetails).flatMap(service -> Try
+            .of(() -> serviceConfigProvider.getConfig(service))
+            .toValidation()
+            .mapError(Throwable::getMessage)
+        )
+            .filter(item -> !com.google.common.base.Strings.isNullOrEmpty(item.getUpdateUrl()))
+            .getOrElseThrow(() -> new CallbackException("Update URL is not configured")).get();
+    }
```

Comment:
maybe it would be clearer to simply check whether case contains service name and **then** get url for it,
instead of doing flatmaps, converting trys to validations etc

Reply:
looks like this will throw exception with message saying `"Update URL is not configured"` when service name does not have case type id

Reply:
we indeed can simplify this method. first of all we need to amend the signature of already existing one so to re-use `service` retrieved prior all these steps:

```java
attachCaseByCcdId(
    event.exceptionRecordJurisdiction,
    targetCaseCcdId,
    event.exceptionRecordDocuments,
    event.exceptionRecordId,
    event.idamToken,
    event.userId,
    exceptionRecordData,
    ignoreWarnings
);
```

`event` already has `service` retrieved from case type id. perhaps we could just pass entire `event` to that method? then `getServiceConfig` gets really simple:

```java
ServiceConfigItem configItem = serviceConfigProvider.getConfig(service);

if (Strings.isNullOrEmpty(item.getUpdateUrl())) {
    throw new CallbackException();
}

return configItem;
```

NOTE. We can replace currently wrong class imported of `Strings` and just use guava's one

Reply:
This validation has already been done, so we shouldn't repeat it. Instead, just retrieve the service config, without any exception handling.